### ISSUE TITLE
incomplete messages parsed as if they were correct?

### ISSUE THREAD

**Issue by horacimacias** [ByteHeathen/libsip#22]
I'm trying to use this parser but I'm seeing some strange behaviour on some messages I believe should not parse correctly (and they do).
For example, if I pass the following text (this is simulating a partially read message):

```
INVITE sip:bob@biloxi.example.com SIP/2.0
Via: SIP/2.0/TCP client.atlanta.example.com:5060;branch=z9hG4bK74bf9
Max-For
```

this ends up being parsed correctly and the parser believes "Max-For" is the body, which is incorrect.

Debug output of the parsed message is

```
Request { method: Invite, uri: Uri { schema: Some(Sip), host: Domain("biloxi.example.com", None), auth: Some(UriAuth { username: "bob", password: None }), parameters: [] }, version: Version(2, 0), headers: Headers([Via(ViaHeader { version: Version(2, 0), transport: Tcp, uri: Uri { schema: None, host: Domain("client.atlanta.example.com", Some(5060)), auth: None, parameters: [Branch("z9hG4bK74bf9")] } })]), body: [77, 97, 120, 45, 70, 111, 114] } []
```

shouldn't the parser expect a double line break (CRLF) after all headers, then optionally read body if Content-Length was found before?

As far as I can tell, this sample message should not parse correctly.

**Comment by bytebuddha** [comment]
There was a error in the parsing login, When parsing the via header it did not check for the CRLF, this was fixed in 08aaaaa18fe719b5cdd533d360ff6fc0bd02dc6

**Comment by horacimacias** [comment]
thanks!
